### PR TITLE
node-manta: 5.3.2 -> 5.4.1

### PR DESCRIPTION
**Pull request by teutat3s**
## Description of changes

https://github.com/TritonDataCenter/node-manta/blob/master/CHANGES.md#540
https://github.com/TritonDataCenter/node-manta/blob/master/CHANGES.md#541

Removed vendored `package-lock.json`, upstream included it in their repo.

Removed update script in favor of `nix-update`.

Adresses https://github.com/NixOS/nixpkgs/pull/230158#issuecomment-1712978336.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/admin/manta/default.nix [hunk 1]
```diff
@@ -1,41 +1,35 @@
...
-  source = lib.importJSON ./source.json;
-in
 buildNpmPackage rec {
-  pname = "manta";
-  inherit (source) version;
-
-  src = fetchurl {
-    url = "https://registry.npmjs.org/${pname}/-/${source.filename}";
-    hash = source.integrity;
+  pname = "node-manta";
```
Comment (dotlambda) [inline pkgs/tools/admin/manta/default.nix:11]:
```suggestion
  pname = "manta";
```
Pnames don't need to be unique.

File: pkgs/tools/admin/manta/default.nix [hunk 2]
```diff
@@ -1,41 +1,35 @@
...
+  version = "5.4.1";
+
+  src = fetchFromGitHub {
+    owner = "TritonDataCenter";
+    repo = "node-manta";
+    rev = "v5.4.1";
```
Comment (dotlambda) [inline pkgs/tools/admin/manta/default.nix:17]:
```suggestion
    rev = "v${version}";
```

File: pkgs/tools/admin/manta/default.nix [hunk 3]
```diff
@@ -1,41 +1,35 @@
...
+    hash = "sha256-C6O5yTCBABMsz2scot8v3IwPbdYvJyZbcPOLfeDXCoo=";
   };
 
-  npmDepsHash = source.deps;
+  npmDepsHash = "sha256-4Zz9sSUXE2dXdkIka2z5bQ2pNmCXXCBS2Sr0JHQOBQw=";
 
   dontBuild = true;
 
   nativeBuildInputs = [ nodejs installShellFiles ];
```
Comment (dotlambda) [inline pkgs/tools/admin/manta/default.nix:25]:
```suggestion
  nativeBuildInputs = [ installShellFiles ];
```

File: pkgs/tools/admin/manta/default.nix [hunk 4]
```diff
@@ -1,41 +1,35 @@
 { lib
 , buildNpmPackage
-, fetchurl
+, fetchFromGitHub
 , nodejs
```
Comment (dotlambda) [inline pkgs/tools/admin/manta/default.nix:4]:
```suggestion
```

**Review by dotlambda — COMMENTED** [review]
Is the update script still needed? `nix-update` should work out of the box

File: pkgs/tools/admin/manta/default.nix (hunk 1 above)
Comment (teutat3s) [reply pkgs/tools/admin/manta/default.nix:11]: With `pname = "manta"`, borg skipped `passthru.tests.version`. Do you know how to keep `pname = "manta"` and tell ofborg to use `node-manta`?

Comment (dotlambda) [reply pkgs/tools/admin/manta/default.nix:10]: @teutat3s Can you try to change the pname just so we can see the ofborg failure?

Comment (cole-h) [reply pkgs/tools/admin/manta/default.nix:10]: The issue is probably that `manta` is an alias to `pkgs.node-manta`, and ofborg skips aliases: https://github.com/NixOS/nixpkgs/blob/479646999c8cf4f7580d294b36742060fe1de170/pkgs/development/node-packages/aliases.nix#L91

Comment (cole-h) [reply pkgs/tools/admin/manta/default.nix:10]: That, or it's related to the fact that there is a `manta` package already: https://github.com/NixOS/nixpkgs/blob/479646999c8cf4f7580d294b36742060fe1de170/pkgs/top-level/all-packages.nix#L39283

So ofborg noticed that `manta` changed and tried to build `manta`, not `node-manta`, as this package is defined as: https://github.com/NixOS/nixpkgs/blob/479646999c8cf4f7580d294b36742060fe1de170/pkgs/top-level/all-packages.nix#L14063

Comment (dotlambda) [reply pkgs/tools/admin/manta/default.nix:10]: But `pkgs.manta` didn't change, only `pkgs.node-manta` did. And that is the package mentioned in the commit message.
@teutat3s You have to use `node-manta` in the commit message regardless of what the pname is.

Comment (teutat3s) [reply pkgs/tools/admin/manta/default.nix:10]: Thank you, that makes sense. I first had `manta` in the commit message, that of course confused borg.

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**